pkg/config/ext_authz.go

```diff
@@ -53,11 +53,32 @@ type ExternalAuthzConfig struct {
 	HTTPFilter *hcm.HttpFilter
 }
 
+type extAuthzProtocol string
+
+const (
+	grpc  extAuthzProtocol = "grpc"
+	http  extAuthzProtocol = "http"
+	https extAuthzProtocol = "https"
+)
+
+var extAuthzProtocols = []extAuthzProtocol{grpc, http, https}
```

Comment:
I think we should use `map`.  Gateway API also uses map like https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/validation/gateway.go#L33-L38